2_base_and_sensor.ipynb

```diff
@@ -172,40 +316,78 @@
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "壁にぶつかってしまいました。上の■ボタンを押して動作を停止させましょう。\n",
     "![title](./imgs/2_stop_button.png)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Oops, the robot bumped into the wall. Press the ■ button above to stop our code.\n",
+    "![title](./imgs/2_stop_button.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "一定時間動かし続けてみましょう。以下を実行すると3秒間直進させることができます。"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Let's keep it running for a limited period of time. Following code will let the robot go straight for 3 seconds:"
+   ]
+  },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "outputs": [],
    "source": [
-    "start_time = get_current_time_sec()  # 現在時刻を取得\n",
-    "while get_current_time_sec() - start_time < 3:  # 3秒経過後ループを抜ける\n",
+    "start_time = get_current_time_sec()  # 現在時刻を取得 Get the current time\n",
+    "while get_current_time_sec() - start_time < 3:  # 3秒経過後ループを抜ける Exit the loop after 3 seconds\n",
     "    move_base_vel(0.5, 0, 0)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "ja"
+   },
    "source": [
     "0.5m/sの速度で3秒間直進させたので、約1.5m前方に進みます。"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "The robot went straight for 3 seconds at the speed of 0.5m/s, so it went forward for about 1.5m."
```

Comment:
"The robot went forward for 3 seconds at the speed of 0.5m/s, so it went forward for about 1.5m."